### DISCUSSION TITLE
“直接计算子图节点数量”和“子图全量点列表长度”不一致，是什么原因呢？

### DISCUSSION THREAD

**Discussion by zjarles** [alibaba/GraphScope#3226] [Q&A] [answered]
sub_node_num = interactive_sub.execute("g.V().count()").one() #直接计算子图节点的数量
print('子图全量点数量',sub_node_num)
node = interactive_sub.execute("g.V()").one()   #将所有节点存成变量，再打印变量长度，应该就是所有节点的数量。 
                                                                           #[v[****1],v[****2,v[****3]...]
print('子图全量点列表长度',len(node))

子图全量点数量 [1290]
子图全量点长度 64

这两个变量为什么不一致呢？

**Comment by shirly121** [comment, marked as answer]
你好，`interactive_sub.execute("g.V()").one()`看起来只会返回某一个点的值，因为调用了`one()`，可以改写成`all()`，就可以返回包含所有点的list
